File: Java/4-PizzaStoreFactory/src/main/java/com/mycompany/pizzastorefactory/ChicagoStylePizza/ChicagoPizzaStore.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.pizzastorefactory.ChicagoStylePizza;

import com.mycompany.pizzastorefactory.NewYorkPizzaStore.NYStyleCheesePizza;
import com.mycompany.pizzastorefactory.Pizza;
import com.mycompany.pizzastorefactory.PizzaStore;

/**
 *
 * @author ismas
 */
public class ChicagoPizzaStore extends PizzaStore {
    protected Pizza createPizza( String item ){
        switch(item){
            case "cheese":
                return new ChicagoStyleCheesePizza();
            default:
                return null;
        }
    }
}

File: C#/4. PizzaStore/Classes/Chicago Pizza Store/ChicagoPizzaStore.cs
using PizzaStore.Classes.New_York_Pizza_Store;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PizzaStore.Classes.Chicago_Pizza_Store
{
    public class ChicagoPizzaStore: PizzaStore
    {
        protected override Pizza CreatePizza(string item)
        {
            switch (item)
            {
                case "cheese":
                    return new ChicagoStyleCheesePizza();
                case "veggie":
                    return new ChicagoStyleVeggiePizza();
                case "clam":
                    return new ChicagoStyleClamPizza();
                case "pepperoni":
                    return new ChicagoStylePepperoniPizza();
                default:
                    return null;
            }
        }
    }
}
